### PR TITLE
[5.2] update to multibyte string functions

### PR DESCRIPTION
**Pull request by hz61p1**

### CODE REVIEW

File: src/Illuminate/Support/Str.php [hunk 1]
```diff
@@ -87,7 +87,7 @@ public static function contains($haystack, $needles)
     public static function endsWith($haystack, $needles)
     {
         foreach ((array) $needles as $needle) {
-            if ((string) $needle === substr($haystack, -strlen($needle))) {
+            if ((string) $needle === mb_substr($haystack, -mb_strlen($needle))) {
```
Comment (mark86092) [inline src/Illuminate/Support/Str.php]: You can use static method
mb_substr --> static::substr
mb_strlen --> static::length

**Comment by GrahamCampbell** [comment]
Maybe this should go to 5.1. Looks like a bug fix to me?

**Comment by mark86092** [comment]
Any test to show the PR fixes bugs?

**Comment by taylorotwell** [comment]
Agree we would probably want this on 5.1 if you don't mind.

**Pull request closed without merging**